cypress/integration/pages/frontPage/tests.js

```diff
@@ -168,42 +194,25 @@ export const testsThatFollowSmokeTestConfig = ({ service, pageType }) =>
             },
           );
         });
-        it('should contain Useful Links if usefulLinks block data exists', () => {
+
+        it('should contain Useful Links if valid usefulLinks block data exists', () => {
           cy.request(`${config[service].pageTypes.frontPage.path}.json`).then(
             ({ body }) => {
               const pageData = body.content.groups;
-              const usefulLinks = pageData.find(data => {
-                return data.type === 'useful-links';
-              });
+              if (shouldDisplayUsefulLinks(pageData)) {
+                // We include the hasStrapline in shouldDisplayUsefulLinks
+                // as we don't render Useful Links if we don't receive a
+                // strapline in the data
+                cy.get('div[data-e2e=useful-links]').should('be.visible');
 
-              const contentTypes = [
-                'Text',
-                'Feature',
-                'Audio',
-                'Video',
-                'Gallery',
-                'Guide',
-              ];
-
-              let usefulLinksItems = [];
-              if (usefulLinks) {
-                usefulLinksItems = usefulLinks.items.filter(item => {
-                  return (
-                    item.assetTypeCode === 'PRO' &&
-                    contentTypes.includes(item.contentType)
-                  );
-                });
-              }
-
-              const isValidUsefulLinks =
-                usefulLinks &&
-                'strapline' in usefulLinks &&
-                usefulLinksItems.length;
-
-              if (isValidUsefulLinks) {
-                //  We include the hasStrapline as we don't render Useful Links
-                //  if we don't receive a strapline in the data
-                cy.get('[aria-labelledby="Useful-links"]').should('be.visible');
+                cy.get('div[data-e2e=useful-links]')
```

Comment:
Do we need to check for `data-e2e=useful-links`? Can't we just check for the `data-e2e="useful-link-item"` if the page should display useful links?

Reply:
Yes, we can just check for the `useful-link-item`. Updated test to check for the item only

Reply:
With this, we will be adding a data attribute to each useful link item and I am not sure if it is the best approach. I would rather add it just to the section and then look within it that there are link items, as we do with the `Index Alsos`. Happy to hear other opinions on this.

Reply:
Opted to keep `data-e2e="useful-link-item"` as I figured it would be an easy approach to check the items. Understand the concern though. Changed to using an attribute on the wrapper and looking within it for useful link `<a>'s`